File: Bubble Sort/Bubble Sort.pas
// BUBBLE SORT IN PASCAL

program Bubble_Sort;

type
	arrayOfInteger = array of integer;

// function algorithm that returns the sorted array
// algoritmo da função que retorna o array ordenado
function bubbleSort(arr: arrayOfInteger): arrayOfInteger; // arrayOfInteger was set in "type" // arrayOfInteger foi definido em "type"
var
	i, j, swap: integer;
begin
	for i := 0 to length(arr) - 2 do
	begin
		for j := i + 1 to length(arr) - 1 do
		begin
			if arr[i] > arr[j] then
			begin
				swap := arr[i];
				arr[i] := arr[j];
				arr[j] := swap;
			end;
		end;
	end;

	bubbleSort := arr;
end;

// code only to call and test the sort function
// código apenas para chamar e testar a função de ordenação
var
	arr: array[0..9] of integer = (5, 2, -3, 10, 23, 99, -1, 7, 93, 0);
	arrSorted: array of integer;
	a: integer;
begin

	arrSorted := bubbleSort(arr);

	writeln('Array Sorted');

	for a in arrSorted do
		writeln(a);
end.

File: Bubble Sort/Bubble Sort.R
#BUBBLE SORT IN R

# function algorithm that returns the sorted array
# algoritmo da função que retorna o array ordenado
bubbleSort <- function(arr)
{
	for (i in 1: (length(arr) - 1))
	{
		for (j in (i + 1): length(arr))
		{
			if (arr[i] > arr[j])
			{
				swap = arr[i]
				arr[i] = arr[j]
				arr[j] = swap
			}
		}
	}

	return (arr)
}

# code only to call and test the sort function
# código apenas para chamar e testar a função de ordenação
arr <- c(5, 2, -3, 10, 23, 99, -1, 7, 93, 0)
arrSorted = bubbleSort(arr)
print("Array Sorted:")

for (i in 1: length(arrSorted))
{
	print(arrSorted[i])
}